packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -110,30 +98,103 @@ library LibProposing {
         // If we choose to persist all data fields in the metadata, it will
         // require additional storage slots.
         unchecked {
-            uint256 rand = uint256(blobHash)
-                ^ (block.prevrandao * b.numBlocks * block.number);
-
             meta = TaikoData.BlockMetadata({
                 l1Hash: blockhash(block.number - 1),
-                // Following the Merge, the L1 mixHash incorporates the
-                // prevrandao value from the beacon chain. Given the possibility
-                // of multiple Taiko blocks being proposed within a single
-                // Ethereum block, we must introduce a salt to this random
-                // number as the L2 mixHash.
-                difficulty: bytes32(rand),
-                blobHash: blobHash,
+                difficulty: 0, // to be initialized below
+                blobHash: 0, // to be initialized below
                 extraData: params.extraData,
                 depositsHash: keccak256(abi.encode(depositsProcessed)),
                 coinbase: msg.sender,
                 id: b.numBlocks,
                 gasLimit: config.blockMaxGasLimit,
                 timestamp: uint64(block.timestamp),
                 l1Height: uint64(block.number - 1),
-                minTier: ITierProvider(resolver.resolve("tier_provider", false))
-                    .getMinTier(rand),
+                txListByteOffset: 0, // to be initialized below
+                txListByteSize: 0, // to be initialized below
+                minTier: 0, // to be initialized below
                 blobUsed: txList.length == 0
             });
         }
+
+        // Following the Merge, the L1 mixHash incorporates the
+        // prevrandao value from the beacon chain. Given the possibility
+        // of multiple Taiko blocks being proposed within a single
+        // Ethereum block, we must introduce a salt to this random
+        // number as the L2 mixHash.
+        unchecked {
+            meta.difficulty = meta.blobHash
```

Comment:
Seems `meta.blobHash` is still uninitialized at this point, so this "salt" is not working?

Reply:
You are right, fixed. Please take another look.